src/consensus/aft/test/driver.h

```diff
@@ -1155,4 +1095,46 @@ class RaftDriver
         _nodes.at(node_id).raft->get_committed_seqno()));
     }
   }
+
+  void assert_detail(
+    ccf::NodeId node_id,
+    const std::string& detail,
+    const std::string& expected,
+    bool equal,
+    const size_t lineno)
+  {
+    auto details = _nodes.at(node_id).raft->get_details();
+    nlohmann::json d = details;
+    if (d.find(detail) == d.end())
+    {
+      RAFT_DRIVER_PRINT(
+        "  Note over {}: Node does not have detail {}", node_id, detail);
+      throw std::runtime_error(fmt::format(
+        "Node {} does not have detail {} on line {}",
+        node_id,
+        detail,
+        std::to_string((int)lineno)));
+    }
+
+    std::string value = d[detail];
+    if (equal ? value != expected : value == expected)
```

Comment:
```suggestion
    if (equal ? (value != expected) : (value == expected))
```

Nit: Odd place for a ternary, so I think this is a big help to readability.

Reply:
Ok as a one-off, but otherwise the clang-format is the law.